client/rpc-spec-v2/src/chain_head/chain_head_follow.rs

```diff
@@ -350,58 +344,47 @@ where
 		};
 
 		// Find the parent hash.
-		let Some(first_number) = self.client.number(*first_hash)? else {
-			return Err(SubscriptionManagementError::BlockNumberAbsent)
-		};
-		let Some(parent) = self.client.hash(first_number.saturating_sub(One::one()))? else {
+		let Some(first_header) = self.client.header(*first_hash)? else {
 			return Err(SubscriptionManagementError::BlockHashAbsent)
```

Comment:
super nit: This sounds like the block hash is absent, but its more `HeaderNotFound` or something.

Reply:
That makes sense! I've changed it to `BlockHeaderAbsent` to make it explicit that we are missing the header from the db, thanks!